ratis-server/src/main/java/org/apache/ratis/server/raftlog/segmented/SegmentedRaftLog.java

```diff
@@ -492,6 +492,13 @@ public void syncWithSnapshot(long lastSnapshotIndex) {
     // if the last index in snapshot is larger than the index of the last
     // log entry, we should delete all the log entries and their cache to avoid
     // gaps between log segments.
+
+    // Close open log segment if entries are already included in snapshot
+    LogSegment openSegment = cache.getOpenSegment();
+    if (openSegment != null && openSegment.getEndIndex() <= lastSnapshotIndex) {
+      fileLogWorker.closeLogSegment(openSegment);
+    }
+    cache.clear();
```

Comment:
We can potentially clear closed log segments with endIndex > snapshot index? Maybe we should move it inside the if statement?

Reply:
Done.